lnd.go

```diff
@@ -381,6 +381,11 @@ func Main(cfg *Config, lisCfg ListenerCfg, shutdownChan <-chan struct{}) error {
 		walletInitParams = *params
 		privateWalletPw = walletInitParams.Password
 		publicWalletPw = walletInitParams.Password
+		defer func() {
+			if err := walletInitParams.Loader.UnloadWallet(); err != nil {
+				ltndLog.Errorf("Could not unload wallet: %v", err)
```

Comment:
This way callers of `Main` would not know that there was an error with `UnloadWallet()` so there's no way to act (panic, or display an error message on the UI).

Reply:
You're right, any non-cli client will not get any feedback here.

This is the case for many things inside `Main` (and things that it calls), only the RPC calls get error feedback really.
For example, the healthchecker currently does play well with the mobile builds: https://github.com/lightningnetwork/lnd/blob/bc6e52888763765793dfa8887e7d88b517a6b913/server.go#L1268-L1313#

I suggest we take a look at mobile feedback issues at a later point. This is something I think is important and want to address, but it think it has to be designed to work with all the different errors `Main` can receive and breaking changes for the mobile build is most likely needed.

Reply:
Yeah, sadly many defer code paths in `Main` ignore returned errors. These error cases should really be refined in the near future to allow safe restarts every time. Fortunately `UnloadWallet` can only fail if it's not loaded which we can ignore and if the DB cannot be closed which is unlikely. So while not ideal, I think we can postpone this refactor for now.

Reply:
Agreed, I'm definitely interested in helping addressing this.